### PR TITLE
dinterpret.d: refactor scrub code

### PR DESCRIPTION
**Pull request by WalterBright**
Refactors `scrubReturnValue()` and `scrubCacheValue()` to:
1. better comments
2. use nested functions to reduce scope of functions
3. eliminate redundant code by using nested functions

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by RazvanN7** [comment]
This PR introduced a regression : https://issues.dlang.org/show_bug.cgi?id=19778

File: src/dmd/dinterpret.d [hunk 1]
```diff
@@ -6464,98 +6525,59 @@ private Expression scrubReturnValue(const ref Loc loc, Expression e)
...
-        // It can be NULL for performance reasons,
-        // see StructLiteralExp::interpret().
-        if (!e)
-            continue;
-
-        // A struct .init may contain void members.
-        // Static array members are a weird special case (bug 10994).
-        if (structlit &&
-            ((e.op == TOK.void_) ||
-             (e.op == TOK.arrayLiteral && e.type.ty == Tsarray && isEntirelyVoid((cast(ArrayLiteralExp)e).elements)) ||
```
Comment (RazvanN7) [inline src/dmd/dinterpret.d]: The `e.type.ty == Tsarray` is omitted in the `isVoid` function making the condition in scrubArray invalid.

Comment (WalterBright) [reply src/dmd/dinterpret.d]: Want to fix it?

Comment (RazvanN7) [reply src/dmd/dinterpret.d]: https://github.com/dlang/dmd/pull/9547

**Pull request merged**